packages/back-end/src/models/ProjectModel.ts

```diff
@@ -1,139 +1,79 @@
-import mongoose from "mongoose";
-import uniqid from "uniqid";
 import { DEFAULT_STATS_ENGINE } from "shared/constants";
-import { omit } from "lodash";
+import { z } from "zod";
 import { ApiProject } from "../../types/openapi";
 import { ProjectInterface, ProjectSettings } from "../../types/project";
-import { ReqContext } from "../../types/organization";
-import { ApiReqContext } from "../../types/api";
+import { statsEngines } from "../util/constants";
+import { baseSchema, MakeModelClass } from "./BaseModel";
 
-const projectSchema = new mongoose.Schema({
-  id: {
-    type: String,
-    unique: true,
-  },
-  organization: {
-    type: String,
-    index: true,
+const projectValidator = baseSchema
+  .extend({
+    name: z.string(),
+    description: z.string(),
+    settings: z.object({
+      statsEngine: z.enum(statsEngines),
+    }),
+  })
+  .strict();
+
+const BaseClass = MakeModelClass({
+  schema: projectValidator,
+  collectionName: "projects",
+  idPrefix: "prj__",
```

Comment:
This should have a single underscore.  The double underscore is unique to fact metrics and not used anywhere else in the app.